Library/Ark/ArkPackage.cs

```diff
@@ -344,30 +346,55 @@ private string[] readFileNames(Stream header)
     }
 
     /// <summary>
-    /// Reads the new file table format in v10
+    /// Reads the new file table format in v9 and v10
     /// </summary>
-    private void readNewFileTable(Stream header)
+    private void readNewFileTable(Stream header, bool readHash = false)
     {
       uint numFiles = header.ReadUInt32LE();
       var files = new OffsetFile[numFiles];
 
-      for (var i = 0; i < numFiles; i++)
+      if (readHash)
       {
-        // Version 3 uses 32-bit file offsets
-        long arkFileOffset = header.ReadInt64LE();
-        string path = header.ReadLengthPrefixedString(System.Text.Encoding.UTF8);
-        var flags = header.ReadInt32LE();
-        uint size = header.ReadUInt32LE();
+        for (var i = 0; i < numFiles; i++)
+        {
+          // Version 3 uses 32-bit file offsets
+          long arkFileOffset = header.ReadInt64LE();
+          string path = header.ReadLengthPrefixedString(System.Text.Encoding.UTF8);
+          var flags = header.ReadInt32LE();
+          uint size = header.ReadUInt32LE();
+          header.ReadUInt32LE();
 
-        var finalSlash = path.LastIndexOf('/');
-        var fileDir = path.Substring(0, finalSlash < 0 ? 0 : finalSlash);
-        var fileName = path.Substring(finalSlash < 0 ? 0 : (finalSlash + 1));
-        var parent = makeOrGetDir(fileDir);
-        var file = new OffsetFile(fileName, parent, contentFileMeta, arkFileOffset, size);
-        file.ExtendedInfo["id"] = i;
-        file.ExtendedInfo["flags"] = flags;
-        files[i] = file;
-        parent.AddFile(file);
+          var finalSlash = path.LastIndexOf('/');
+          var fileDir = path.Substring(0, finalSlash < 0 ? 0 : finalSlash);
+          var fileName = path.Substring(finalSlash < 0 ? 0 : (finalSlash + 1));
+          var parent = makeOrGetDir(fileDir);
+          var file = new OffsetFile(fileName, parent, contentFileMeta, arkFileOffset, size);
+          file.ExtendedInfo["id"] = i;
+          file.ExtendedInfo["flags"] = flags;
+          files[i] = file;
+          parent.AddFile(file);
+        }
+      }
+      else
+      {
+        for (var i = 0; i < numFiles; i++)
```

Comment:
Since this loop is almost the same as the one above, except for skipping the checksum, can you just move the `readHash` check to within the loop? I'd like to try to minimize duplicate code.

Reply:
I did it this way so there'd only be one comparison instead of x comparisons (equal to file count). But I guess that's over optimization for our purpose.